### DISCUSSION TITLE
Inquiry: Single-band grayscale GeoTIFF

### DISCUSSION THREAD

**Discussion by Saijin-Naib** [deckerst/aves#515] [Q&A]
I have found Aves by looking for something that was FOSS and can handle raster data like GeoTIFF. It works beautifully in general and has become my Gallery app, but I'm now just noticing that my Single-band GeoTIFF files (DEM products mainly) can not be rendered.

Is this something in-scope of Aves?

**Comment by deckerst** [comment]
Could you share some sample files?
If it's possible I'd be happy to make it work, but no promise.

**Reply by Saijin-Naib** [reply]
Thanks for taking a look! Aves is absolutely incredible, and I'd like to recommend it for Android users who dabble in geospatial work as nothing else exists that comes close!

These are representative of pretty much this entire class of GIS raster data (odm_dem folder):  
https://hub.dronedb.app/r/saijinnaib/brighton-beach-road-06-23-2016-all

There are also other slightly more esoteric ones, like paletted/classified:  
https://github.com/Saijin-Naib/QGIS-ESRI-Testfiles/tree/trunk/GeoTIFF

If there is any other test data or anything else you'd like or need, please just let me know.

**Reply by deckerst** [reply]
Thanks, I'll take a look when I can!

By the way, I tried to bring some GeoTIFF related features:
- deriving the image location from the GeoTIFF data when there is no Exif GPS section,
- overlaying the GeoTIFF on the map.

But these features are currently undertested and incomplete. I hope to refine them in the future.

**Reply by deckerst** [reply]
Taking for example `odm_dem/dsm.tif`, I see a sensible gradient when displayed in DroneDB gallery. But the file doesn't display properly in my usual desktop image software (XnViewMP, Gimp, etc.). Do you know any desktop/mobile software that display these files correctly?

**Reply by Saijin-Naib** [reply]
About the only thing that displays it moderately correctly is Affinity Photo, which is closed-source.
![image](https://user-images.githubusercontent.com/19295950/217698382-7b3f72ea-2027-4d7e-acf3-a3789c4bc9d3.png)
GIS software like QGIS obviously load it just fine... Perhaps the solution would have to be to load a colormap when a single-band image is attempted to be loaded to map the values to in a linear fashion, much like how a GIS software would handle it. The Virdis color ramp is widely regarded as the optimal one in this space.
